library/src/main/java/com/novoda/downloadmanager/MigrationServiceBinder.java

```diff
@@ -8,31 +8,42 @@
 
 public class MigrationServiceBinder {
 
-    private final Context context;
-    private final Migrator.Callback migrationCallback;
+    private final Context applicationContext;
+    private final Callback migrationCallback;
+    private final NotificationChannelCreator notificationChannelCreator;
+    private final NotificationCreator<MigrationStatus> notificationCreator;
 
-    private MigrationServiceConnection serviceConnection;
+    public interface Callback {
```

Comment:
❓ QUESTION ❓ 

Had to expose a `Callback` that the clients use to receive update events from the `Migrator`. Could do this another way by exposing the `Migrator` to the client. Or we could expose just the listener. I was unsure, what do you think?

Reply:
Maybe if it's a top level class called `MigrationCallback` or spoilers ahead for my review... using `Promises`